docs: improve contribution guides

tiagolobocastro opened the pull request:
## Pull Request template

**Why is this PR required? What issue does it fix?**:
I had a bit of trouble running the tests and getting them to pass recently.
We can improve things.

**What this PR does?**:
Improves on the existing documentation.
Facilitates testing with a no-frills ready-made vm. 

**Does this PR require any upgrade changes?**:
No

**Checklist:**
- [ ] Fixes #<issue number>
- [x] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [ ] Has the change log section been updated?
- [ ] Commit has unit tests
- [ ] Commit has integration tests
- [ ] (Optional) Are upgrade changes included in this PR? If not, mention the issue/PR to track:
- [ ] (Optional) If documentation changes are required, which issue on https://github.com/openebs/openebs-docs is used to track them:

tiagolobocastro committed:
test: add nixos shell

Adds a nixos shell for local testing.
This a headless vm with lvm installed and configured for testing,
as well as being a single node k3s cluster.

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

tiagolobocastro committed:
docs: enhance contributors development setup guide

Lint contributing doc
Improve development setup guide, including the nixos-shell k3s docs

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

tiagolobocastro committed:
test: improve ux with cli arguments

Use cli arguments to control cleanup rather than using
env variables which are more obscure.

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

tiagolobocastro committed:
build: disable cgo

Rather than add static flags and use musl, do the same as zfs
and disable cgo

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

dsharma-dc commented on docs/developer-setup.md:
```diff
@@ -125,38 +131,179 @@ Always start with creating a new branch from develop to work on a new feature or
+
+###### MiniKube
+
+To install minikube follow the doc [here](https://kubernetes.io/docs/tasks/tools/install-minikube/).
```
Since this is a developer setup - shall we mention specifically that `driver` should be chosen as `none` during minikube start. Otherwise, if any local image is wanted that doesn't get picked up. This was something we ran into one I remember @Abhinandan-Purkait ?

tiagolobocastro replied:
Yes good catch, it has to be none as it stands today, or otherwise we need another step such as this one: https://github.com/openebs/lvm-localpv/blob/b3059ef75f7d4577fea8ed013e78b20eb29efe8b/ci/ci-test.sh#L231

tiagolobocastro replied:
Done

tiagolobocastro committed:
test: use test role label

Add test role label for easier cleanup
Simplify jq command on cleanup
Also improves test duration by using retry loop rather than hardcoded sleep.

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>
